config/docker/fragment/kernelci.jinja2

```diff
@@ -12,8 +12,7 @@ RUN git clone --depth=1 $core_url /tmp/kernelci-core
 WORKDIR /tmp/kernelci-core
 RUN git fetch origin $core_rev
 RUN git checkout FETCH_HEAD
-RUN pip3 install -r requirements-dev.txt
-RUN python3 setup.py install
+RUN pip3 install .[dev]
```

Comment:
Is this equivalent to `pip3 install -r requirements-dev.txt`?  If so, I think it's a bit less explicit and I'm not sure of the advantage.

Also I don't get why the `setup.py install` command was removed, but then I don't remember why it was added either.  Could you please shed some light on this?

Reply:
The commit message says:

> Make use of pip instead of calling setup.py directly.

but it doesn't give the reason for doing this.  Is it fixing or improving anything?

Reply:
It's not the equivalent  to `pip3 install -r requirements-dev.txt`. It's rather the equivalent of

```
pip3 install -r requirements-dev.txt
python3 setup.py install
```

I actually find using `setup.py` and `requrements-dev.txt` a bit confusing. 
Why should we install the application with `setup.py` and some additional packages from `requirements-dev.txt` when there is a mechanism for specifying required and extra packages in `setup.py`?

Reply:
I think the key issue here was about installing the pinned-down package versions in requirements.txt.  Now I'm not sure if we should drop requirements.txt and specify the versions in setup.py, although I believe it's really the norm to only specify pinned down versions in requirements.txt.

Reply:
setup.py and requirements.txt serve a bit different purpose. The setup.py file is more like a package recipe in Linux distributions. It doesn't necessarily need to specify the exact version of the dependencies if it's not absolutely needed to have a specific version to have package up and running.  This is why it's more common to have no version specified (getting the newest available)  or just specify the minimum and/or maximum version of a dependency. 
Basically running `setup.py install` should give us similar effects as running e.g. `apt install` for Debian packages.

So, if kernelci is supposed to run with any version of the packages listed in `install_requires` then there is no need to specify those.

requirements.txt is more of an "environment snapshot", which is why there are usually listed specific versions of the packages.

In my opinion we should use setup.py/pip  to install the kernelci package which should result in having the kernelci package installed in the container together with all its dependencies required for proper run.

Reply:
I think we agree on this point as it's the established norm, the versions should be in requirements.txt as a snapshot of a working set of dependencies.

Then it's clear that installing requirements.txt with pip installs the versioned packages from the file.  And it's also clear that we need to use `setup.py` to install the kernelci-core package from the git clone.  What's not so clear is what `setup.py` will do with the dependencies since they're already installed using `pip`, and my guess is that it doesn't do anything since they're already installed.

Therefore, doing `pip install -r requirements-dev.txt` for the dependencies and then `python3 setup.py install` for kernelci-core makes complete sense in this case to me.  What's the reason for changing this again?

Reply:
If `pip3 install .[dev]` is really equivalent to doing the 2-step install with `pip` and then `setup.py`, then it's not a functional change but just a syntaxic thing.  I'd rather keep the explicit commands unless the shorter one is really more widespread but I'm not that familiar with it.  But please confirm whether it's actually equivalent or if it does anything different.

Reply:
Having versioned packages means we're in control of when packages get updated, which is very important for the production images.  If we want to depart from that then we can discuss it but it's a larger topic so I guess it shouldn't be part of this PR which is about adding Docker version tags.